File: 283-MoveZeroes/283-MoveZeroes.cpp
// Last updated: 8/13/2025, 8:20:41 PM
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n=nums.size();
        vector<int> answer;
        int s=0;;
        for(int x:nums){
            if(x!=0){
                nums[s]=x;
                s++;
            }
                
        }
        for(int i=s;i<n;i++)
            nums[i]=0;
            
                
        }
    
    
};

File: 283-MoveZeroes/283-MoveZeroes.js
// Last updated: 8/13/2025, 8:20:28 PM
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let i = 0;
    for(let j = 0;j<nums.length;j++){
        if(nums[j]!==0){
            [nums[i] , nums[j]] = [nums[j] , nums[i]];
            i++;
        }
    }
};